back-end/routes/api/users.js

```diff
@@ -29,16 +29,20 @@ router.post("/register", async (req, res) => {
   }
 
   // Try to find an existing user
-  let user = await User.findOne({ email: req.body.email });
+  let user = await User.findOne({ email: req.body.email, username: req.body.email });
```

Comment:
Here, I think you will actually want two separate queries - one to check if there is a user with that email, and one to check if there is a user with that username. Something like

```js
let userWithEmail = await User.findOne({ email: req.body.email });
let userWithUsername = await User.findOne({ username: req.body.username });
```

Reply:
Yep, the current logic will only find an existing user that has both the provided username *and* email, which is too strict and would actually allow someone to register using an existing username or email, as long as the *other* one doesn't match.

Reply:
This has been updated 🙂